lib.go

```diff
@@ -192,9 +191,12 @@ func getFieldByTag(ival interface{}, tagName string, fieldNames []string) (inter
 
 		parts := strings.Split(tagValue, ",")
 		if parts[0] == fieldName {
-			value := val.Field(i).Interface()
+			value := val.Field(i)
 			if len(rest) == 0 {
-				return value, nil
+				if value.CanInterface() {
+					return value.Interface(), nil
+				}
+				return nil, trace.BadParameter("field %v is not accessible", fieldName)
```

Comment:
Should this be a hard error or just a `notFoundError`? Other similar problematic operations (accessing a subfield of a non-struct) seem to just return `notFoundError`.

```suggestion
				return nil, &notFoundError{fieldNames: fieldNames}
```

Reply:
https://github.com/gravitational/predicate/pull/21